test/service/test_response.py

```diff
@@ -1945,7 +1945,36 @@ def test_filter_by_publication_title(self):
                 }
                 self.assertEqual(expected_files, files)
 
+    def test_access(self):
+        filtered_entity_types = ['bundles', 'files', 'samples']
+        unfiltered_entity_types = ['projects']
```

Comment:
```suggestion
        filtered_entity_types = {
            'bundles': True, 
            'files': True,
            'samples': True,
            'projects': False
        }
```

is the normal form (your version allows a type to occur in both lists, and more than once in each list) and looks like it will also simplify the code below.